### ISSUE TITLE
Login to pre-amazon account fails

### ISSUE THREAD

**Issue by biuklija** [rmcrackan/Libation#163]
I've been unsuccessfully trying to log in into two Audible UK pre-Amazon accounts. Tried both login types (username/password and browser login), both report errors on library sync.

```
2021-11-30 13:47:52.453 +01:00 [INF] (at AppScaffolding.LibationScaffolding.logStartupState(LibationFileManager.Configuration)) Begin. {"AppName":"Libation","Version":"6.5.3.2","Mode":"Release","LogLevel_Verbose_Enabled":false,"LogLevel_Debug_Enabled":false,"LogLevel_Information_Enabled":true,"LogLevel_Warning_Enabled":true,"LogLevel_Error_Enabled":true,"LogLevel_Fatal_Enabled":true,"LibationFiles":"C:\\Users\\XXXXX\\Libation","BooksDirectory":"d:\\Libation\\LibationFiles\\Books","InProgress":"C:\\Users\\XXXXX\\AppData\\Local\\Temp\\Libation","DownloadsInProgressDirectory":"C:\\Users\\XXXXX\\AppData\\Local\\Temp\\Libation\\DownloadsInProgress","DownloadsInProgressFiles":0,"DecryptInProgressDirectory":"C:\\Users\\XXXXX\\AppData\\Local\\Temp\\Libation\\DecryptInProgress","DecryptInProgressFiles":0}
2021-11-30 13:47:55.911 +01:00 [INF] (at ApplicationServices.LibraryCommands.GetCounts()) Book counts. {"total":1224,"booksFullyBackedUp":14,"booksDownloadedOnly":0,"booksNoProgress":1210,"booksError":0}
2021-11-30 13:47:55.924 +01:00 [INF] (at ApplicationServices.LibraryCommands.GetCounts()) PDF counts. {"total":7,"pdfsDownloaded":0,"pdfsNotDownloaded":7}
2021-11-30 13:48:03.760 +01:00 [INF] (at AudibleUtilities.ApiExtended+<CreateAsync>d__5.MoveNext()) {"LoginType":"ILoginChoiceEager","Account":"AccountId=XXXXX|AccountName=s[...]K|Locale=pre-amazon - uk","LocaleName":"pre-amazon - uk"}
2021-11-30 13:48:17.128 +01:00 [INF] (at LibationWinForms.Dialogs.Login.LoginChoiceEagerDialog.submitBtn_Click(System.Object, System.EventArgs)) Submit button clicked: {"email":"XXXXX","passwordLength":8}
2021-11-30 13:48:18.597 +01:00 [INF] (at AudibleApi.Authentication.LoginResultRunner+<makeRequestAsync>d__6.MoveNext()) Send request {"method":"POST","uri":"https://www.audible.co.uk/ap/signin","absoluteUri":"[== uri]"}
2021-11-30 13:48:19.260 +01:00 [INF] (at AudibleApi.EzApiCreator+<loginEmailPasswordAsync>d__6.MoveNext()) Login result: "AudibleApi.Authentication.CaptchaPage"
2021-11-30 13:48:27.224 +01:00 [INF] (at LibationWinForms.Dialogs.Login.CaptchaDialog.submitBtn_Click(System.Object, System.EventArgs)) Submit button clicked: {"Answer":"4rswm7"}
2021-11-30 13:48:27.237 +01:00 [INF] (at AudibleApi.Authentication.LoginResultRunner+<makeRequestAsync>d__6.MoveNext()) Send request {"method":"POST","uri":"https://www.audible.co.uk/ap/signin","absoluteUri":"[== uri]"}
2021-11-30 13:48:28.059 +01:00 [INF] (at AudibleApi.EzApiCreator+<loginEmailPasswordAsync>d__6.MoveNext()) Login result: "AudibleApi.Authentication.LoginComplete"
2021-11-30 13:48:28.594 +01:00 [ERR] (at ApplicationServices.LibraryCommands+<ImportAccountAsync>d__1.MoveNext()) Error importing library
AudibleApi.Authorization.RegistrationException: Could not register
 ---> System.Net.Http.HttpRequestException: Response status code does not indicate success: 500 (Internal Server Error).
   at System.Net.Http.HttpResponseMessage.EnsureSuccessStatusCode()
   at AudibleApi.Authorization.Authorize.RegisterAsync(AccessToken accessToken, IEnumerable`1 cookies)
   --- End of inner exception stack trace ---
   at AudibleApi.Authorization.Authorize.RegisterAsync(AccessToken accessToken, IEnumerable`1 cookies)
   at AudibleApi.Authorization.IdentityMaintainer.RegisterAsync()
   at AudibleApi.Authorization.IdentityMaintainer.CreateAsync(IIdentity identity, IAuthorize authorize, ISystemDateTime systemDateTime)
   at AudibleApi.Authorization.IdentityMaintainer.CreateAsync(IIdentity identity)
   at AudibleApi.EzApiCreator.createApiAsync(IIdentity identity)
   at AudibleApi.EzApiCreator.createApiAsync(Identity inMemoryIdentity, String identityFilePath, String jsonPath)
   at AudibleApi.EzApiCreator.GetApiAsync(ILoginChoiceEager loginChoiceEager, Locale locale, String identityFilePath, String jsonPath)
   at AudibleUtilities.ApiExtended.CreateAsync(Account account, ILoginChoiceEager loginChoiceEager) in C:\Dropbox\DinahsFolder\coding\_NET\Visual Studio 2022\Libation\AudibleUtilities\ApiExtended.cs:line 30
   at ApplicationServices.LibraryCommands.scanAccountsAsync(Func`2 apiExtendedfunc, Account[] accounts, ResponseGroupOptions libraryResponseGroups) in C:\Dropbox\DinahsFolder\coding\_NET\Visual Studio 2022\Libation\ApplicationServices\LibraryCommands.cs:line 136
   at ApplicationServices.LibraryCommands.ImportAccountAsync(Func`2 apiExtendedfunc, Account[] accounts) in C:\Dropbox\DinahsFolder\coding\_NET\Visual Studio 2022\Libation\ApplicationServices\LibraryCommands.cs:line 86
2021-11-30 13:48:28.623 +01:00 [ERR] (at LibationWinForms.MessageBoxAlertAdmin.Show(System.String, System.String, System.Exception)) Alert admin error: {"text":"Error importing library. Please try again. If this still happens after 2 or 3 tries, stop and contact administrator","caption":"Error importing library"}
AudibleApi.Authorization.RegistrationException: Could not register
 ---> System.Net.Http.HttpRequestException: Response status code does not indicate success: 500 (Internal Server Error).
   at System.Net.Http.HttpResponseMessage.EnsureSuccessStatusCode()
   at AudibleApi.Authorization.Authorize.RegisterAsync(AccessToken accessToken, IEnumerable`1 cookies)
   --- End of inner exception stack trace ---
   at AudibleApi.Authorization.Authorize.RegisterAsync(AccessToken accessToken, IEnumerable`1 cookies)
   at AudibleApi.Authorization.IdentityMaintainer.RegisterAsync()
   at AudibleApi.Authorization.IdentityMaintainer.CreateAsync(IIdentity identity, IAuthorize authorize, ISystemDateTime systemDateTime)
   at AudibleApi.Authorization.IdentityMaintainer.CreateAsync(IIdentity identity)
   at AudibleApi.EzApiCreator.createApiAsync(IIdentity identity)
   at AudibleApi.EzApiCreator.createApiAsync(Identity inMemoryIdentity, String identityFilePath, String jsonPath)
   at AudibleApi.EzApiCreator.GetApiAsync(ILoginChoiceEager loginChoiceEager, Locale locale, String identityFilePath, String jsonPath)
   at AudibleUtilities.ApiExtended.CreateAsync(Account account, ILoginChoiceEager loginChoiceEager) in C:\Dropbox\DinahsFolder\coding\_NET\Visual Studio 2022\Libation\AudibleUtilities\ApiExtended.cs:line 30
   at ApplicationServices.LibraryCommands.scanAccountsAsync(Func`2 apiExtendedfunc, Account[] accounts, ResponseGroupOptions libraryResponseGroups) in C:\Dropbox\DinahsFolder\coding\_NET\Visual Studio 2022\Libation\ApplicationServices\LibraryCommands.cs:line 136
   at ApplicationServices.LibraryCommands.ImportAccountAsync(Func`2 apiExtendedfunc, Account[] accounts) in C:\Dropbox\DinahsFolder\coding\_NET\Visual Studio 2022\Libation\ApplicationServices\LibraryCommands.cs:line 86
   at LibationWinForms.Dialogs.IndexLibraryDialog.IndexLibraryDialog_Shown(Object sender, EventArgs e) in C:\Dropbox\DinahsFolder\coding\_NET\Visual Studio 2022\Libation\LibationWinForms\Dialogs\IndexLibraryDialog.cs:line 34

```

**Comment by rmcrackan** [comment]
Thanks for the log snippet. Which error is it from: in-app? Or in browser?

**Comment by biuklija** [comment]
This particular one is in-app, but if you need the browser one, I have that too. Fails equally.

**Comment by rmcrackan** [comment]
Yes, please. I'd like to take a look at that too.

Expectation setting: I'm about at the limits of my ability for the amazon/audible login stuff. What exists now is a long evolution of improvements and I'm running out of ideas. I'll try to resolve this but no promises.

**Comment by rmcrackan** [comment]
Fixed in latest version [v6.6](https://github.com/rmcrackan/Libation/releases/latest)
